src/main/test/CommandHandlerTests.cpp

```diff
@@ -137,3 +145,447 @@ TEST_CASE("transaction envelope bridge", "[commandhandler]")
+            auto txFrame = root.tx({dataOp});
+            submitTx(txFrame->getEnvelope(), retStr);
+
+            {
+                LedgerTxn lookupTx(app->getLedgerTxnRoot());
+                auto entry = lookupTx.load(LedgerEntryKey(dle));
+                REQUIRE(!entry);
+            }
+
+            submitClose(noLedgerSeq, noCloseTime, retStr);
+
+            LedgerTxn lookupTx(app->getLedgerTxnRoot());
+            auto entry = lookupTx.load(LedgerEntryKey(dle));
+            REQUIRE(entry);
+        }
+
+        SECTION("A transaction which is submitted on time can become late by "
+                "the time of a ledger close")
+        {
+            LedgerEntry dle;
+            dle.lastModifiedLedgerSeq = lastLedgerNum();
+            dle.data.type(DATA);
+            auto de = LedgerTestUtils::generateValidDataEntry();
+            de.accountID = root.getPublicKey();
+            dle.data.data() = de;
+
+            auto dataOp = txtest::manageData(de.dataName, &de.dataValue);
+            auto txFrame = root.tx({dataOp});
+            REQUIRE(txFrame->getEnvelope().type() == stellar::ENVELOPE_TYPE_TX);
+            txFrame->getEnvelope().v1().tx.timeBounds.activate();
+            txFrame->getEnvelope().v1().tx.timeBounds->minTime = 0;
+            TimePoint const maxTime =
+                lastCloseTime() + defaultManualCloseTimeInterval +
+                getUpperBoundCloseTimeOffset(*app, lastCloseTime());
+            txFrame->getEnvelope().v1().tx.timeBounds->maxTime = maxTime;
+            txFrame->getEnvelope().v1().signatures.clear();
+            txFrame->addSignature(root);
+
+            {
+                LedgerTxn checkLtx(app->getLedgerTxnRoot());
+                auto valid = txFrame->checkValid(checkLtx, 0, 0, 0);
+                REQUIRE(valid);
+            }
+
+            submitTx(txFrame->getEnvelope(), retStr);
+
+            {
+                LedgerTxn lookupTx(app->getLedgerTxnRoot());
+                auto entry = lookupTx.load(LedgerEntryKey(dle));
+                REQUIRE(!entry);
+            }
+
+            SECTION("Ledger closes on time; transaction is applied")
+            {
+                submitClose(noLedgerSeq, make_optional<TimePoint>(maxTime),
+                            retStr);
+
+                LedgerTxn lookupTx(app->getLedgerTxnRoot());
+                auto entry = lookupTx.load(LedgerEntryKey(dle));
+                REQUIRE(entry);
+            }
+
+            SECTION("Ledger closes too late; transaction is not applied")
+            {
+                submitClose(noLedgerSeq, make_optional<TimePoint>(maxTime + 1),
+                            retStr);
+
+                LedgerTxn lookupTx(app->getLedgerTxnRoot());
+                auto entry = lookupTx.load(LedgerEntryKey(dle));
+                REQUIRE(!entry);
+            }
+        }
+
+        SECTION("Large transaction queues may be emptied over the course of "
```

Comment:
this test is getting way too deep into testing "everything that core does", we already have tests for the transaction queue, and this is just duplicating that so there is no real added test coverage, on the other hand we're making our test suite slower.

Reply:
Okay, I've removed it.